smarts/core/route_cache.py

```diff
@@ -51,6 +51,8 @@ class RouteWithCache(RoadMap.Route):
     def __init__(self, road_map: RoadMap):
         self._map = road_map
         self._logger = logging.getLogger(self.__class__.__name__)
+        self._start_lane: Optional[RoadMap.Lane] = None
```

Comment:
Should the new attributes affect the `hash`?

Reply:
`start_lane` and `end_lane` is now included in the hash.